app/src/main/java/com/google/android/gms/samples/wallet/PaymentsUtil.java

```diff
@@ -53,129 +55,227 @@ private PaymentsUtil() {
-            paramsBuilder.addParameter(param.first, param.second);
-        }
+    private static JSONObject getGatewayTokenizationSpecification() throws JSONException {
+        JSONObject tokenizationSpecification = new JSONObject();
 
-        return createPaymentDataRequest(transactionInfo, paramsBuilder.build());
+        tokenizationSpecification.put("type", "PAYMENT_GATEWAY");
+        JSONObject parameters = new JSONObject(Constants.GATEWAY_TOKENIZATION_PARAMETERS);
+        tokenizationSpecification.put("parameters", parameters);
+
+        return tokenizationSpecification;
     }
 
     /**
-     * Builds {@link PaymentDataRequest} for use with DIRECT integration to be consumed by
-     * {@link PaymentsClient#loadPaymentData}.
-     * <p>
-     * Please refer to the documentation for more information about DIRECT integration. The type of
+     * Direct Integration: Decrypt a response directly on your servers. This configuration has
+     * additional data security requirements from Google and additional PCI DSS compliance complexity.
+     *
+     * <p>Please refer to the documentation for more information about DIRECT integration. The type of
      * integration you use depends on your payment processor.
      *
-     * @param transactionInfo contains the price for this transaction.
+     * @return Payment data tokenization for the CARD payment method.
+     * @throws JSONException
+     * @see <a
+     *     href="https://developers.google.com/pay/api/android/reference/object#PaymentMethodTokenizationSpecification">PaymentMethodTokenizationSpecification</a>
      */
-    public static PaymentDataRequest createPaymentDataRequestDirect(TransactionInfo transactionInfo) {
-        PaymentMethodTokenizationParameters params =
-                PaymentMethodTokenizationParameters.newBuilder()
-                        .setPaymentMethodTokenizationType(
-                                WalletConstants.PAYMENT_METHOD_TOKENIZATION_TYPE_DIRECT)
+    private static JSONObject getDirectTokenizationSpecification() throws JSONException {
 
+        JSONObject tokenizationSpecification = new JSONObject();
+        tokenizationSpecification.put("type", "DIRECT");
+        tokenizationSpecification.put(
+                "parameters",
+                new JSONObject()
+                        .put("protocolVersion", "ECv1")
                         // Omitting the publicKey will result in a request for unencrypted data.
-                        // Please refer to the documentation for more information on unencrypted
-                        // requests.
-                        .addParameter("publicKey", Constants.DIRECT_TOKENIZATION_PUBLIC_KEY)
-                        .build();
+                        // Please refer to the documentation for more information on unencrypted requests.
+                        .put("publicKey", Constants.DIRECT_TOKENIZATION_PUBLIC_KEY));
```

Comment:
check if DIRECT_TOKENIZATION_PUBLIC_KEY is set or is default at start of method. reject early